src/coreclr/jit/importer.cpp

```diff
@@ -7256,6 +7256,13 @@ void Compiler::impCheckForPInvokeCall(
 
     if (suppressGCTransition)
     {
+        if (opts.compMCJ)
+        {
+            // MultiCoreJit thread can't inline PInvoke with SuppressGCTransitionAttribute,
```

Comment:
It is less than idea to be hardcoding policy like this in the JIT. I think it would be better to implement this policy on the VM side in the `pInvokeMarshalingRequired` method instead.

Reply:
What about passing explicit flag that tells jit to skip inlining for pinvokes with SuppressGCTransitionAttribute in mcj thread like in https://github.com/gbalykov/runtime/commit/124bcaf9063d07fd7799c4e0d59e4095714396c6?

`pInvokeMarshalingRequired` approach will require change with additional arg passing, because on vm side we don't know whether this is mcj thread compilation or not:
```c++
bool CEEInfo::pInvokeMarshalingRequired(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* callSiteSig, bool isMCJ)
```

Reply:
If it's useful it is possible to get the `PrepareCodeConfig` for the in-progress compilation in callbacks to the VM like this:
https://github.com/dotnet/runtime/blob/4bab77944c38e4c80af23b90ee24133fa1ef6c0a/src/coreclr/vm/jitinterface.cpp#L6910-L6912

Reply:
@kouvel thanks a lot, that's a piece of logic I was looking for! I've updated PR.